pkg/kubectl/cmd/apply.go

```diff
@@ -220,7 +230,13 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *App
 			helper := resource.NewHelper(info.Client, info.Mapping)
 			patcher := NewPatcher(encoder, decoder, info.Mapping, helper, overwrite)
 
-			patchBytes, err := patcher.patch(info.Object, modified, info.Source, info.Namespace, info.Name)
+			infos := []*resource.Info{info}
```

Comment:
Hmmm... I'll check, but from what I see [kubectl edit](https://github.com/kubernetes/kubernetes/blob/0006d6611edb095d84220cc2b5f5b5d3d7558b54/pkg/kubectl/cmd/edit.go#L159) is using the same approach.